pkg/dynamicinstrumentation/ditypes/config.go

```diff
@@ -172,12 +174,16 @@ func (pi *ProcessInfo) CloseConfigUprobe() error {
 // SetUprobeLink associates the uprobe link with the specified probe
 // in the tracked process
 func (pi *ProcessInfo) SetUprobeLink(probeID ProbeID, l *link.Link) {
+	pi.mu.Lock()
+	defer pi.mu.Unlock()
 	pi.InstrumentationUprobes[probeID] = l
 }
 
 // CloseUprobeLink closes the probe and deletes the link for the probe
 // in the tracked process
 func (pi *ProcessInfo) CloseUprobeLink(probeID ProbeID) error {
+	pi.mu.Lock()
```

Comment:
We could probably utilize a [`sync.Map`](https://pkg.go.dev/sync#Map) here to ensure `InstrumentationUprobes` is thread-safe. Does the use-case fit the criteria? 

> The sync.Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys

Reply:
I see the value in using `sync.Map` for simplicity and broader thread-safety beyond this source file. I’ve noted this for future consideration but opted for a simple mutex since adding/removing probes in InstrumentationUprobes is infrequent.
In the meantime, I’ve ensured additional protections in 08bbb5ca0e3b60bb114299e210867816b8274a4f.